engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/ValidationContext.scala

```diff
@@ -26,11 +30,15 @@ case class ValidationContext(localVariables: Map[String, TypingResult] = Map.emp
   def contains(name: String): Boolean = variables.contains(name)
 
   def withVariable(name: String, value: TypingResult)
-                  (implicit nodeId: NodeId): ValidatedNel[PartSubGraphCompilationError, ValidationContext] =
-    if (variables.contains(name))
+                  (implicit nodeId: NodeId, paramName: Option[String] = None): ValidatedNel[PartSubGraphCompilationError, ValidationContext] = {
+    if (variables.contains(name)) {
       Invalid(OverwrittenVariable(name)).toValidatedNel
-    else
+    } else if (!VariableNamePattern.matcher(name).matches()) {
```

Comment:
I think this else if is misleading, let's make those validations independently and then just combine errors... (I know that if variable is defined, it name will be valid, just for sake of readability)

Reply:
You're right - done.